src/pages/enterprise/auth-connect/azure-ad.md

```diff
@@ -0,0 +1,116 @@
+
+It's recommended to create an `AuthenticationService` class that encapsulates Azure AD and Ionic Auth Connect’s login functionality.
+
+Generate this class using the `ionic generate` command:
+
+<command-line>
+<command-prompt>ionic generate service services/authentication</command-prompt>
+</command-line>
+
+Extend the `IonicAuth` class, then configure all Azure AD details in the `IonicAuthOptions` object:
+
+```typescript
+import { IonicAuth, IonicAuthOptions } from '@ionic-enterprise/auth';
+
+export class AuthenticationService extends IonicAuth {
+
+constructor() {
+    const azureConfig: IonicAuthOptions = {
+        // The auth provider.
+        authConfig: 'azure',
+        // The platform which the app is running on
+        platform: 'cordova',
+        // client or application id for provider
+        clientID: 'FILL_IN',
+        // the discovery url for the provider
+        // OpenID configuration
+        discoveryUrl: 'FILL_IN',
+        // the URI to redirect to after log in
+        redirectUri: 'FILL_IN',
+        // requested scopes from provider
+        scope: 'FILL_IN',
+        // the audience, if applicable
+        audience: 'FILL_IN',
+        // the URL to redirect to after log out
+        logoutUrl: 'FILL_IN',
+        // The type of iOS webview to use. 'shared' will use a webview that can 
+        // share session/cookies on iOS to provide SSO across multiple apps but
+        // will cause a prompt for the user which asks them to confirm they want
+        // to share site data with the app. 'private' uses a webview which will
+        // not prompt the user but will not be able to share session/cookie data
+        // either for true SSO across multiple apps.
+        iosWebView: 'private',
+        // required if running on the Web
+        clientSecret: ''
+    };
+    
+    super(auth0Config);
+    }
+}
+```
+
+Some of these `IonicAuthOptions` values are unique, and must be set based on your Azure AD app’s details:
+
+* `platform`: Use “cordova” or “capacitor” accordingly.
+* `clientID`: Your app’s _Application ID_. Example: cebbb0be-d578-4bbd-9712-4b0fe05c06aa
+* `redirectUri`: The URI to redirect to after the user has logged in. Use the same AUTH_URL_SCHEME variable value (App Id) from when the Auth Connect plugin was installed. Example: com.company.app://callback
+* `logoutUrl`: The URI to redirect to after the user has logged out. Example: com.company.app://login?logout=true
+* `audience`: Your custom API URL of choice, such as `https://api.myapp.com`.
+* `scope`: Unlock access to protected resources, such as read/write permissions. Example: openid offline_access email picture profile
+
+The `discoveryUrl` consists of the following format:
+
+`https://B2C-TENANT-NAME.b2clogin.com/B2C-TENANT-NAME.onmicrosoft.com/v2.0/.well-known/openid-configuration?p=POLICY-NAME`
+
+Replace `B2C-TENANT-NAME` with your tenant name and the `POLICY-NAME` with the name of the User Flow created earlier.
```

Comment:
Did I explain this correctly? Got the explanation from [here](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-add-identity-providers#create-an-azure-active-directory-application).

Reply:
the other place to find it is on the azure portal:
azure ad b2c-> app -> user flows (policies) -> run user flow -> the discovery link is at the top

also for non-b2c is is a bit easier.